functions/src/common/datastore.ts

```diff
@@ -15,110 +15,111 @@
+    const doc = await docRef.get();
+    return (doc.exists ? doc.data() : null);
   }
 
-  fetchDoc_(path) {
+  fetchDoc_(path: string) {
     return this.fetch_(this.db_.doc(path));
   }
 
-  fetchCollection_(path) {
+  fetchCollection_(path: string) {
     return this.db_.collection(path).get();
   }
 
-  fetchSurvey(surveyId) {
+  fetchSurvey(surveyId: string) {
     return this.db_.doc(`surveys/${surveyId}`).get();
   }
 
-  fetchRecord(surveyId, loiId, recordId) {
+  fetchRecord(surveyId: string, loiId: string, recordId: string) {
     return this.fetchDoc_(
       `surveys/${surveyId}/lois/${loiId}/records/${recordId}`
     );
   }
 
-  fetchSubmissionsByJobId(surveyId, jobId) {
+  fetchSubmissionsByJobId(surveyId: string, jobId: string) {
     return this.db_
       .collection(`surveys/${surveyId}/submissions`)
       .where("jobId", "==", jobId)
       .get();
   }
 
-  fetchLocationOfInterest(surveyId, loiId) {
+  fetchLocationOfInterest(surveyId: string, loiId: string) {
     return this.fetchDoc_(`surveys/${surveyId}/lois/${loiId}`);
   }
 
-  fetchLocationsOfInterestByJobId(surveyId, jobId) {
+  fetchLocationsOfInterestByJobId(surveyId: string, jobId: string) {
     return this.db_
       .collection(`surveys/${surveyId}/lois`)
       .where("jobId", "==", jobId)
       .get();
   }
 
-  fetchTask(surveyId, loiTypeId, taskId) {
+  fetchTask(surveyId: string, loiTypeId: string, taskId: string) {
     return this.fetchDoc_(
       `surveys/${surveyId}/loiTypes/${loiTypeId}/tasks/${taskId}`
     );
   }
 
-  fetchSheetsConfig(surveyId) {
+  fetchSheetsConfig(surveyId: string) {
     return this.fetchDoc_(`surveys/${surveyId}/sheets/config`);
   }
 
-  async insertLocationOfInterest(surveyId, loi) {
+  async insertLocationOfInterest(surveyId: string, loi: any) {
     const loiDoc = {
       ...loi,
       geometry: Datastore.toFirestoreMap(loi.geometry),
     };
-    const docRef = await this.db_.collection("surveys").doc(surveyId);
+    const docRef = this.db_.collection("surveys").doc(surveyId);
     const doc = await docRef.get();
     if (!doc.exists) {
       throw new Error(`/surveys/${surveyId} not found`);
     }
     await docRef.collection("lois").add(loiDoc);
   }
-}
 
-function toFirestoreValue(value) {
-  if (value === null) {
-    return null;
-  }
-  if (Array.isArray(value)) {
-    if (value.length === 2 && value.every((x) => typeof x === "number")) {
-      return new firestore.GeoPoint(value[0], value[1]);
+  static toFirestoreMap(geometry: any) {
+    return Object.fromEntries(
+      Object.entries(geometry).map(([key, value]) => [
+        key,
+        Datastore.toFirestoreValue(value),
+      ])
+    );
+  };
+
+  static toFirestoreValue(value: any): any {
+    if (value === null) {
+      return null;
+    }
+    if (Array.isArray(value)) {
+      if (value.length === 2 && value.every((x) => typeof x === "number")) {
+        return new firestore.GeoPoint(value[0], value[1]);
```

Comment:
In a subsequent commit I reversed the order of x and y since GeoJSON coords are reversed. I'll send a PR with both of our commits shortly.